src/rules/no-flowmax-in-forwardref.coffee

```diff
@@ -0,0 +1,23 @@
+module.exports =
+  meta:
+    docs:
+      description: 'Flag uses of flowMax() within forwardRef()'
+      category: 'Possible Errors'
+      recommended: yes
+    schema: [
+      type: 'object'
+      additionalProperties: no
+    ]
+    fixable: 'code'
+
+  create: (context) ->
+    report = (node) ->
+      context.report {
+        node,
+        message: "flowMax() cannot be used within forwardRef()"
+      }
+
+    "CallExpression[callee.name='forwardRef'] CallExpression[callee.name='flowMax']":
```

Comment:
Wow I wasn't familiar with how powerful these "extended selectors" were - can you point me at the docs or wherever you read about this?

It looks like these are working correctly, that's quite a savings (code-wise) compared to eg manually keeping track of whether you're inside a `forwardRef()` call/sniffing nodes with helpers like `isFlowMax()`

Reply:
Here are the selector docs: https://eslint.org/docs/developer-guide/selectors. I came across them because I was looking for a way to see if a node had any descendants matching some criteria without having to manually walk through the tree. It looked like in the other rules in this plugin they only need to look one level deep, avoiding that issue. I'm still not sure how exactly I'd have done that if these selectors didn't exist—I'd be curious to hear how you'd have implemented this rule.

Reply:
I would've kept a flag (either a single boolean or a stack) indicating whether we're currently inside a `flowMax()` call eg
```
inFlowMax = no

CallExpression: (node) ->
  inFlowMax = yes if isFlowMax node

  report node if inFlowMax and isForwardRef node

'CallExpression:exit': (node) ->
  inFlowMax = no if isFlowMax node
```
But you can see from that why you'd typically use a stack for tracking it since if you had nested `flowMax()`'s a single boolean flag would be insufficient

So more like:
```
inFlowMax = []

CallExpression: (node) ->
  inFlowMax.push {} if isFlowMax node

  report node if inFlowMax.length and isForwardRef node

'CallExpression:exit': (node) ->
  inFlowMax.pop() if isFlowMax node
```